internal/controller/controllers/clusterdeployments_controller.go

```diff
@@ -230,7 +231,7 @@ func (r *ClusterDeploymentsReconciler) Reconcile(origCtx context.Context, req ct
 
 	if swag.StringValue(cluster.Kind) == models.ClusterKindCluster && !clusterInstall.Spec.HoldInstallation {
 		// Day 1
-		pullSecret, err := getAndLabelPullSecret(ctx, r.Client, r.APIReader, clusterDeployment.Spec.PullSecretRef, req.Namespace)
+		pullSecret, err := r.PullSecretClient.GetValidPullSecret(ctx, getPullSecretKey(req.Namespace, clusterDeployment.Spec.PullSecretRef))
```

Comment:
It was added because it needed a state (k8s client and api reader). The creation of the state cleaned the API and allowed it to be re-used from both kube layer and REST

Reply:
ok so worth moving later also getSecret func to handler: https://github.com/openshift/assisted-service/blob/cca573243ca975d0bd4d6c4ec212586804bee833/internal/controller/controllers/common.go#L58

Reply:
It's an internal utility not and API